### ISSUE TITLE
`[Network/VPNGateway]`  deployment operation failed due to an intermittent error

### ISSUE THREAD

**Issue by wuxu92** [Azure/azure-rest-api-specs#21164]
## Brief
VPN gateway deployment keeps failing with helpless information

## Description
When creating a virtual network gateway for VPN Type with route-based policy, the create operation returns with server-side error without debugging/helpful information to investigate. I have tried with different SKU types and modified the customeRoute prefix value,   which makes no different and all failed.

##  related API: 
https://github.com/Azure/azure-rest-api-specs/blob/ae227e20bcb22d83634452026f701d75bec2619e/specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/virtualNetworkGateway.json#L1937

## example

gateway request payload

```json
{
    "location": "eastus",
    "properties": {
        "activeActive": false,
        "customRoutes": {
            "addressPrefixes": [
                "101.168.0.6/32"
            ]
        },
        "enableBgp": false,
        "enablePrivateIpAddress": true,
        "gatewayType": "Vpn",
        "ipConfigurations": [
            {
                "name": "vnetGatewayConfig",
                "properties": {
                    "privateIPAllocationMethod": "Dynamic",
                    "publicIPAddress": {
                        "id": "/subscriptions/xxx-xxxx/resourceGroups/xxx-rg/providers/Microsoft.Network/publicIPAddresses/vpnpubip001"
                    },
                    "subnet": {
                        "id": "/subscriptions/xxx-xxxx/resourceGroups/xxx-rg/providers/Microsoft.Network/virtualNetworks/vpngw001/subnets/GatewaySubnet"
                    }
                }
            }
        ],
        "sku": {
            "name": "VpnGw3AZ",
            "tier": "VpnGw3AZ"
        },
        "vpnType": "RouteBased"
    }
}
```

reponse error message:

```json
{
    "error": {
        "code": "VmssGatewayDeploymentFailed",
        "details": [],
        "message": "The gateway deployment operation failed due to an intermittent error. Please try again."
    },
    "status": "Failed"
}
```

**Comment by mayank-reynencourt** [comment]
Hi,

i'm also trying to create VNG and face same issue , any update on this ?

**Comment by el-memer** [comment]
Hi, 
I'm also facing same issue here (Region West Europe), I've opened a support ticket and will send updates here if I have some.

**Comment by birdnathan** [comment]
Same here - West Europe. Please fix!

**Comment by FletchAD** [comment]
Same here - UKSouth and UKWest, tried with VpnGw1 & VpnGw1AZ, interestingly the basic SKU seems to work ok, but that's not what I need.

**Comment by slaffka-vlasov** [comment]
The same is for uswest3 with VpnGw1 & VpnGw2. Basic doesn't work as well

**Comment by FrankMormino** [comment]
Same thing here - for west Europe as well.

**Comment by FrankMormino** [comment]
> Hi, I'm also facing same issue here (Region West Europe), I've opened a support ticket and will send updates here if I have some.

Same here - trying the same in West Europe - any luck on the support ticket side?

**Comment by whosfj** [comment]
Same issue here. Happening in West Europe trying to deploy a VpnGw2AZ.

**Comment by BevanSin** [comment]
Note there is an issue currently with Azure Key Vault that may be causing this problem - if you look in the Service Health page there is an alert there that is affecting downstream Azure services like VPN Gateway.

**Comment by birdnathan** [comment]
[Image.png]
________________________________

**Comment by birdnathan** [comment]
From Azure support:
“Hi Nathan
 
Thank you for contacting Microsoft Azure Networking support.
 
Please know that we are currently investigating an ongoing global issue regarding Virtual Network Gateway (VPN Gateway) deployments and we will get back to you with updates once this has been mitigated.”

Seems the key vault issue is wiping out network services. The status website need to reflect this and not just list key vault as impacted

**Comment by haciz** [comment]
Hello, 

Indeed joining the list to report an issue, neither the empty PUT request is helping

`Set-AzVirtualNetworkGateway: Long running operation failed with status 'Failed'. Additional Info:'The gateway deployment operation failed due to an intermittent error. Please try again.'
StatusCode: 200
ReasonPhrase: OK
Status: Failed
ErrorCode: VmssGatewayDeploymentFailed
ErrorMessage: The gateway deployment operation failed due to an intermittent error. Please try again.`

**Comment by imkevinjones** [comment]
Same issue here... This has been happening all day.

**Comment by shanyuen** [comment]
Yesterday I bought a subscription and created a fresh virtual network in southeast asia.
But still error.

Status: Conflict
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"VmssGatewayDeploymentFailed","message":"The gateway deployment operation failed due to an intermittent error. Please try again."}]}
SKU: VpnGw1
Correlation id: 2f5880ba-a8b2-4184-801e-a5b8cea6f3b4

**Comment by FletchAD** [comment]
After this confirmation at 3rd Nov @ 05:52 UTC https://status.azure.com/en-us/status/history/ I was able to provision VpnGw1 in UKSouth this morning.

**Comment by whosfj** [comment]
Seems to be mitigated. Can also deploy the VGW now.

**Comment by el-memer** [comment]
Got an answer from the support team.

> I apologize for the inconvenience that was caused but as we checked, there is a service disruption in all regions, which impacts the deployments of the VPN Gateway resources. We hope that it might be resolved by the end of the day. Our internal team are actively investigating and working on a solution, however, currently we do not have a specific resolution date or time.
 
I tried to redeploy this morning, hoping it'll be resolved soon.

**Comment by slaffka-vlasov** [comment]
Confirmed creation of VpnGw1 in west us 3

**Comment by laglergruener** [comment]
Hi,
we‘ve currently the same issue, mentioned above in WestEurope for AZ gateways.
Are there some issues still available?
Thanks,
Hannes

**Comment by rik-v** [comment]
This issue seems to be current again. We're currently unable to deploy any VPN gateway to any subscription or any tenant.
Tried 3 different tenants, 4 different subscriptions, 2 types of SKU (VpnGw1 and VpnGw2), all in West Europe.
We noticed this starting january 9, 2024, however, it is not unlikely this issue persisted before that date.

No mentions of this on the service health pages. So, curious as to what's causing it this time.

**Comment by soufianerabi** [comment]
Yeah, for the past three days, we've been facing the same issue in our organization. The deployment has failed multiple times, even after many retries. Unfortunately, Microsoft support wasn't helpful. We're still trying to deploy this resource, but it continues to fail.

**Comment by tanarchytan** [comment]
Also a problem on my side. tried it since 09 of January. Region West Europe with SKU VPNGw2

> {
  "code": "DeploymentFailed",
  "target": "/subscriptions/34303d6e-8fe7-4908-a10d-3013ec206f5f/resourceGroups/rg-net-hub/providers/Microsoft.Resources/deployments/Microsoft.Template-20240115105923",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.",
  "details": [
    {
      "code": "ResourceDeploymentFailure",
      "target": "/subscriptions/34303d6e-8fe7-4908-a10d-3013ec206f5f/resourceGroups/rg-net-hub/providers/Microsoft.Network/virtualNetworkGateways/vpng-derooy-prod",
      "message": "The resource write operation failed to complete successfully, because it reached terminal provisioning state 'Failed'."
    }
  ]
}

>

**Comment by rik-v** [comment]
@soufianerabi @tanarchytan, I tried deploying in north europe last friday (jan. 12th, 2024), which completes successfully, which makes me believe Microsoft simply has some serious resource issues in the west europe region.

It's not the first time (or sign of) this is happening, a few months back we were unable to roll out other resources (VM's, VMSS'es (for AKS)) because apparently the resource was unavailable in a specific zone in west europe.
We ended up deploying to "just" 2 zones instead of 3 (still ok, but we wanted 3 :)).
To be clear: we haven't checked if those resources have become available again in all zones since then, so perhaps things improved (although I strongly doubt that considering the fact we can't deploy VPN to west europe)

Would be helpful if Microsoft would simply clarify the current state of west europe, if there's a capacity issue, that's annoying, but I'd rather have them tell us then having to find out this way...

**Comment by TGosselink** [comment]
Some here, VPN deploy in West Europe:
The gateway deployment operation failed due to an intermittent error. Please try again. (Code: VmssGatewayDeploymentFailed)

**Comment by lgriffithsdoherty** [comment]
Me too

**Comment by foxmeyson** [comment]
Microsoft is not good, there are problems all the time. That's horrible

**Comment by rbnmk** [comment]
Currently facing same issue in West Europe for VpnGw1 and VpnGw1AZ

**Comment by mtc3net** [comment]
Me too

**Comment by lgriffithsdoherty** [comment]
![Screenshot 2024-01-16 143451](https://github.com/Azure/azure-rest-api-specs/assets/146077232/50ec2d8d-ef58-416a-96ae-b12cd1a5853b)

Microsoft have called me to as that they are not sure what the issue is and it could be months before it is resolved. The attached states capacity constraints but still no resolution in site just a warning set up to say capacity constraints wont allow you to build gateways in west europe for the foresable.

**Comment by rik-v** [comment]
@lgriffithsdoherty , thank you for sharing this! :)
This very unfortunate news, especially for new customers :( 
Their proposed (temporary) solution, although understandable, is unlikely to help much though as you can only deploy a gateway to a VNet in the same region. Which means you'll have to move your virtual network too, or you have to start managing 2. Also, this might incur outbound network traffic cost (because resources might be in different regions).

I hope Microsoft makes quick work of expanding their capacity in west europe :/

**Comment by atovivan** [comment]
This is a serious blocking issue for us as well, that will possibly postponed our go-live date. I try to move only the vnet and vpn gateway to a different region but without success, possible we will need to move all network related resources or the entire infrastructure to a different region. For now we are using a bastion VM but that is not a cheap solution as on one VM only two people can work simultaneously.

Looks like Microsoft doesn't care much about this problem.

**Comment by efcorpa** [comment]
Cannot believe this is happening and even that Microsoft says it could be months before it is resolved.
Can anybody confirm it works with a given SKU?
Is there any workaround other than creating a VNet in another region and peer it with the one at WE?

**Comment by redoz** [comment]
@efcorpa  FWIW I tried deploying every SKU multiple times yesterday and it failed every single time.

**Comment by jordyvpaassen** [comment]
@efcorpa I've the same issue by different customers, the only one option is to make a resource group with a vpn gateway and a vnet in a other region and make a peer between your current vnet and the one that is deployed in a other region. If you have questions let me know

**Comment by mestief** [comment]
The problem unfortunately still exists 👎 
Guess I'll have to use North Europe for now.

**Comment by Agger1995** [comment]
I have been retrying every day for a few weeks at this point, and only just today was I able to succesfully deploy a VPN Gateway to westeurope, with the SKU `VpnGW1`.
_Something_ has been fixed, but I can't find any official information on the incident anywhere.

**Comment by redoz** [comment]
> I have been retrying every day for a few weeks at this point, and only just today was I able to succesfully deploy a VPN Gateway to westeurope, with the SKU `VpnGW1`. _Something_ has been fixed, but I can't find any official information on the incident anywhere.

Probably just enough people gave up on westeurope and picked a different region. We moved everything to Sweden central, not being able to reliably deploy things is not an risk we're willing to take.

**Comment by MarkTallentire** [comment]
Over a year later and this issue still persists. At a minimum it'd be nice if it was capacity issues that this is reflected in the error message.

**Comment by MarkTallentire** [comment]
Yes, still seem to be getting this in West Europe and North Europe.

LargePrefPlaceholder-XKUz1MEJBwkOM

**Comment by JacopoNicolosiAlten** [comment]
same problem in West Europe for VPNGw2
